client/my-sites/pages/page-list.jsx

```diff
@@ -461,6 +462,7 @@ export default class PageList extends Component {
 		return (
 			<div>
 				<QueryPosts siteId={ siteId } query={ { ...query, page } } />
+				<QuerySiteSettings siteId={ siteId } />
```

Comment:
Sounds good! It might be better to request site settings after the theme switch 👍

Another way is to let `theme-setup` endpoint return the related site options so we can simply update the local data.

Reply:
Previously I thought about that but then what's the purpose of `QuerySiteSettings`? Why did we create that?

I think the "client" of any site settings / options should be responsible to fetch the latest value of the settings/options that they need. Here, the Pages page needs the latest value of the front page options, so it needs to fetch them using `QuerySiteSettings`.

What do you think?

Reply:
I don't think the way it's implemented will matter that much, so I'm starting to feel that it would be fine to address this at another time. 🙇  But I comment in an idealistic sense.

I think a component like `QuerySiteSettings` is for data encapsulation so that each component can manage its own data in a reliable way.

Perhaps we could consider a hybrid approach: Whenever a setting is updated (like the theme-switching), we could update the Redux store, and also trigger a local data refresh on each component (fetching in the background might be better). In this way, overall efficiency is improved, but components are still responsible for their own data. I think this kind of pattern is often seen in a project using GraphQL.

Reply:
Thanks @arthur791004  & @okmttdhr for your reviews!

After reading your comments and contemplating, I've found a more appropriate place to put this logic. Could you check again? 🙏

Reply:
Looks good to me! Thank you for your time 🚀